test/integration/test_dwarf.c

```diff
@@ -205,305 +146,41 @@ bool test_dwarf3_cpp_basic(void) { // this should work for dwarf2 aswell
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(13);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_const_type);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(2);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_pointer_type);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(3);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_reference_type);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(3);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_subroutine_type);
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(3);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_unspecified_parameters);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(1);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_base_type);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(4);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_pointer_type);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(4);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_structure_type);
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(8);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_inheritance);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(3);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_subprogram);
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(8);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_subprogram);
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(10);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_subprogram);
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(13);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_subprogram);
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(12);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_variable);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(7);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_variable);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(7);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_subprogram);
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(8);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_formal_parameter);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(5);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_subprogram);
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(5);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_formal_parameter);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(4);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_subprogram);
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(9);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_formal_parameter);
-	{
-		check_abbrev_children(false);
-		check_abbrev_count(3);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_subprogram);
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(9);
-	}
-	i++;
-	check_abbrev_tag(DW_TAG_subprogram);
-	{
-		check_abbrev_children(true);
-		check_abbrev_count(8);
-	}
+	da = rz_bin_dwarf_abbrev_parse(bin->cur);
+	mu_assert("Incorrect number of abbreviation", rz_bin_dwarf_abbrev_count(da) == 32);
+
+	RzBinDwarfAbbrevTable *tbl = ht_up_find(da->tbl_by_offset, 0x0, NULL);
+	char *dump = rz_core_bin_dwarf_abbrev_decl_to_string(rz_bin_dwarf_abbrev_get(tbl, 1));
```

Comment:
@imbillow could you please address @ret2libc suggestion and check these properties in the code instead? I believe it's the only major change necessary left.